docs/md_merge.py

```diff
@@ -0,0 +1,357 @@
+import argparse
+import os
+import re
+import shutil
+from argparse import BooleanOptionalAction
+from collections import OrderedDict, defaultdict
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Optional
+
+MD_HEADER_REGEX = (
+    r"^"            # start of line
+    r"(\s{0,3})"    # Group 1: zero to three whitespaces, tabs etc (four and more spaces are no heading)
+    r"(#{1,6})"     # Group 2: one or up to 6 '#'
+    r"(\s{0,})"     # Group 3: zero or more whitespaces
```

Comment:
Not sure if this is a good idea. Markdown headers SHOULD have a single space between the hash(es) and the header text - no space might indicate something not intended as header ... no idea about the use cases but maybe fronting the script with a rewriting markdown linter might be a good idea.

Reply:
Good point, I updated it to require at least one space.

Actually, the use case here implies that the markdown files in question are linted before applying this script.